src/cli.ts

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env node
+
+import { addDays, format, isBefore } from "date-fns";
+import prompts, { Answers } from "prompts";
+import { getConfig, overrideConfig } from "./config";
+import { run } from "./shared/leaderboard";
+import { logError } from "./shared/shared-logger";
+
+// `prompts` package uses generics which accepts string literal values and then uses those values as types
+// This means I have to ensure I use a string literal type that matches a string literal value
+const PROMPT_NAME = "value";
+type PromptType = "value";
+
+const setStartDate = async () => {
+    const promptData: Answers<PromptType> = await prompts({
+        name: PROMPT_NAME,
+        type: "date",
+        message: "Enter a start date: ",
+        initial: new Date(Date.now()),
+        mask: "DD-MM-YYYY",
+    });
+
+    return promptData[PROMPT_NAME];
+};
+
+const setEndDate = async (startDate: Date) => {
+    const datePlus2Weeks = addDays(new Date(startDate), 14);
+
+    const promptData: Answers<PromptType> = await prompts({
+        name: PROMPT_NAME,
+        type: "date",
+        message: "Enter an end date: ",
+        initial: datePlus2Weeks,
+        mask: "DD-MM-YYYY",
+        validate: (endDate) => {
+            if (isBefore(endDate, startDate)) {
+                const startDateString = format(startDate, "dd/MM/yyyy");
+                const endDateString = format(endDate, "dd/MM/yyyy");
+
+                return `endDate (${endDateString}) cannot be earlier than startDate (${startDateString})`;
+            } else {
+                return true;
+            }
+        },
```

Comment:
Extract to `endDateValidator()`.  Maybe this could be reused in the config verifiier?